### PR TITLE
[WIP] allow one hot encoder to handle missing data

### LINKED ISSUE #587: allow One hot encoder to work with nan
At the moment the one hot encoder does not support missing values. It will raise an error. We want to make this transformer more versatile.

OneHotEncoder could have 3 options:

- [ ] missing_values="raise" ==> current functionality
- [ ] missing_values="encode" ==> treat na as an additional category
- [ ] missing_values="ignore" ==> ignore, but only if encoding in k-1, otherwise they will be merged with the dropped category

### PR DESCRIPTION
**Pull request by SunnyxBd**
closes #587

### CODE REVIEW

File: feature_engine/encoding/one_hot.py [hunk 1]
```diff
@@ -210,7 +215,12 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
 
         X = check_X(X)
         variables_ = self._check_or_select_variables(X)
-        _check_contains_na(X, variables_)
+        
+        # check if dataset contains na if missing_value is 'raise' or empty
+        if self.missing_values == "raise":
+            _check_contains_na(X, variables_)
+        elif self.missing_values == "encode":
+            X = X.fillna('novalue_temp')
```
Comment (solegalli) [inline feature_engine/encoding/one_hot.py:223]: This is looking really good :)

Could we use a different string? like"nan" or "missing_value"?

We should probable only fillna in `X[variables_]`

File: feature_engine/encoding/one_hot.py [hunk 2]
```diff
@@ -269,8 +279,11 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
 
         X = self._check_transform_input_and_state(X)
 
-        # check if dataset contains na
-        _check_contains_na(X, self.variables_)
+        # check if dataset contains na if missing_value is 'raise' or empty
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables_)
+        elif self.missing_values == "encode":
+            X = X.fillna('novalue_temp')
```
Comment (solegalli) [inline feature_engine/encoding/one_hot.py:286]: we should probable only fillna in `X[variables_]`

**Comment by SunnyxBd** [comment]
Hi @solegalli 

I'm unfortunately unable to run any of the CircleCI Pipeline.
I created an account with my github credentials. 

I have the following error 
`Orb codecov/codecov@3.2.3 not loaded. To use this orb, an organization admin must opt-in to using third party orbs in Organization Security settings.`

Did any of the contributors encounter this before? Do you have an idea about how to fix this?

**Comment by solegalli** [comment]
Hey @SunnyxBd 

codecov and circleci should run automatically on this PR whenever you commit anything new to your branch. You don't need to add credentials to your github account. So I'd say don't worry about that.

So far, circleci has ran on all other PR's until yesterday. It means nothing, it can break from a minute to the next.

I'd say dont worry about circleci or codecov. If it still does not run, I will look into it in a day or 2. 

If your tests pass locally, just commit them here and I sort circle ci later. Cheers

**Comment by solegalli** [comment]
Make sure your main branch is in sync with the main branch of this repo though. And if it wasn't sync it and then rebase onto your feature branch

**Pull request closed without merging**